tests/test_core/test_mock_all.py

```diff
@@ -18,3 +19,22 @@ def test_multiple_services():
     ddb = boto3.client("dynamodb", region_name=rgn)
     r = ddb.list_tables()
     r["ResponseMetadata"]["HTTPStatusCode"].should.equal(200)
+
+
+def test_context_manager():
+    with mock_all():
+        rgn = "us-east-1"
+        sqs = boto3.client("sqs", region_name=rgn)
+        r = sqs.list_queues()
+        r["ResponseMetadata"]["HTTPStatusCode"].should.equal(200)
+
+        lmbda = boto3.client("lambda", region_name=rgn)
+        r = lmbda.list_event_source_mappings()
+        r["ResponseMetadata"]["HTTPStatusCode"].should.equal(200)
+
+        ddb = boto3.client("dynamodb", region_name=rgn)
+        r = ddb.list_tables()
+        r["ResponseMetadata"]["HTTPStatusCode"].should.equal(200)
+
+    with pytest.raises(Exception):
+        sqs.list_queues()
```

Comment:
ServerMode will patch any clients inside a decorator by setting the `endpoind_url`-attribute.
So when calling this method here, it uses the client that still has the `endpoint_url` set - regardless of whether the mock is active or not.

As clients are cached within a session, the best option here is probably to create a new session, and let that session create a 'clean' client (without the endpoint_url):
```
with mock_all():
    ....

unpatched_sqs = boto3.Session().client("sqs", region_name=rgn)

with pytest.raises(Exception):
    unpatched_sqs.list_queues()
```

That passes in both the unit tests and ServerMode tests